core/templates/pages/exploration-player-page/services/hints-and-solution-manager.service.ts

```diff
@@ -163,7 +163,11 @@ export class HintsAndSolutionManagerService {
   // WARNING: This method has a side-effect. If the retrieved hint is a
   // pending hint that's being viewed, it starts the timer for the next
   // hint.
-  displayHint(index: number): SubtitledHtml | null {
+  displayHint(index: number | undefined): SubtitledHtml | null {
+    if (typeof index !== 'number') {
```

Comment:
In what scenario will index be undefined?

Reply:
Actually, index has been initialised in class DisplayHintModalComponent, and in the ngOnInit() function, value of index has not been set. So, index remains undefined.

Reply:
Can we only call this function with number index?